Project.toml

```diff
@@ -31,8 +31,9 @@ Interpolations = "a98d9a8b-a2ab-59e6-89dd-64a1c18fca59"
 IterativeSolvers = "42fd0dbc-a981-5370-80f2-aaf504508153"
 JLD2 = "033835bb-8acc-5ee8-8aae-3f567f8a3819"
 JuliaFormatter = "98e50ef6-434e-11e9-1051-2b60c6c9e899"
-Kronecker = "2c470bb0-bcc8-11e8-3dad-c9649493f05e"
+Juno = "e5e0dc1b-0480-54bc-9374-aad01c23163d"
 KernelAbstractions = "63c18a36-062a-441e-b654-da1e3ab1ce7c"
+Kronecker = "2c470bb0-bcc8-11e8-3dad-c9649493f05e"
```

Comment:
```suggestion
```
Is this package used? I do not see it in the code

Reply:
It is used in test_derivatives. Going through the dependencies and cleaning up what we do not need would be next after agreeing that we want this PR merged and using the workflow in the script.

Reply:
I suggest that later after the cleanup we start using [`[weakdeps]` and `[extensions]`](https://pkgdocs.julialang.org/v1/creating-packages/#Weak-dependencies)

Reply:
Sure! nice to check what can be the best practices there.

Reply:
I added the link to the best practices in the relevant issue. lets do this later then